### ISSUE TITLE
Columns: Make it easier to add a column from the canvas

### ISSUE THREAD

**Issue by jeryj** [WordPress/gutenberg#67843]
## What problem does this address?

It's too hard to know how to insert a new column after the initial column selection on the columns placeholder. 

In order to add a column, you need to understand:
- That a column is within the parent columns block
- How to select the columns parent block via the list view or parent block selector
- Find the columns number within the block settings

## What is your proposed solution?

Potential ideas:
- Have an "Add Column" option in both the Columns and Column block toolbars. Both would append one more column to the group. 
- Some kind of in-canvas indicator to add a new column on the far left and right of the columns block.

**Comment by jeryj** [comment]
@WordPress/gutenberg-design - Feel free to close if you don't think this is a good route :)

**Comment by Mamaduka** [comment]
@yogeshbhutkar, I think this needs design feedback before we proceed with implementation.

@jeryj, in my experience, the Insert before/after block options work well for Columns + in-between inserter. The main problem with the "Add Column" toolbar button is the block's position; It's unclear where it will be inserted.

cc @WordPress/gutenberg-design

**Comment by yogeshbhutkar** [comment]
Including an "Add Column" button in the toolbar feels more intuitive and user-friendly. Sure, I'll wait for design feedback on this approach.

>  The main problem with the "Add Column" toolbar button is the block's position; It's unclear where it will be inserted.

If triggered from the parent `Columns` block, the new column should intuitively be inserted as the first child. Conversely, if triggered from an individual `Column` block, it makes sense to insert the new column immediately after the currently selected block.

**Comment by Mamaduka** [comment]
IMO, intuition is usually personal. Example: I would expect a new column to be the last child; others expect it to be the first.

The container blocks already provide tools to insert child blocks, so maybe we should improve those instead of providing block-specific solutions.

**Comment by jeryj** [comment]
> IMO, intuition is usually personal. Example: I would expect a new column to be the last child; others expect it to be the first.

I agree with @Mamaduka.

> It's unclear where it will be inserted.

I agree, but it's easier to figure out how to move around a column than it is to figure out how to add a new column. Insert it anywhere, move it afterwards :)

**Comment by jasmussen** [comment]
Valid enough issue, but tricky to find a good solution primarily because: a column is a block, and in general the block inserter is used for adding blocks, not buttons in the toolbar. Counterpoint: the usual user experience when creating _tables_ or _spreadsheets_ is to click "add row" or "add column" buttons in a nearby toolbar. Finally, there's the the project on the whole to consider: there's a lot of UI, a lot of features already, so simply adding more buttons comes with a price. That's not to say we can't do it, but it does suggest, for those pondering various solutions to this, to look at improving existing UI if at all possible:

Is there a way to more intuitively use the block inserter to add columns? Probably not since the _column_ is already suggested:

![Image](https://github.com/user-attachments/assets/f034be9b-a428-41de-9b7c-90909e7061d0)

Is there a way to improve the sibling inserter to have a UI for this? There's been discussion on how to add blocks before or after the first or last blocks in the document, presumably a good solution for that could also work horizontally, i.e. a vertical sibling inserter left of the columns block would add a column before it, and vice versa. 

Is there a way to surface the black inserter button to the right of the columns block when selected, like it is the case for the Navigation block?

![Image](https://github.com/user-attachments/assets/466b6afa-36ae-4616-9eb2-a96f2f8717c3)

These are mainly suggestions for further creative thinking at this point.

**Comment by jeryj** [comment]
@jasmussen - I should have named the issue "Surface way to add columns from the canvas" or something else more general. I agree that putting it in the block toolbar is a divergence, and it would be better if there was a way to surface it in the canvas on the left/right of the columns instead somehow.  

Has there ever been column placeholders if they're empty? Right now there's no canvas indication if there's an empty column. If we have placeholders for empty unselected columns, it may be easier to differentiate "add a block to this column" vs "add a new column".

**Comment by Mamaduka** [comment]
It looks like the inner blocks appender was disabled in https://github.com/WordPress/gutenberg/pull/20502.

> Has there ever been column placeholders if they're empty? Right now there's no canvas indication if there's an empty column.

The editor used to show child block appenders, but now it only displays them for selected blocks. I can't find where changes were introduced, but @jasmussen might remember.

**Comment by jasmussen** [comment]
> The editor used to show child block appenders, but now it only displays them for selected blocks. I can't find where changes were introduced, but @jasmussen might remember.

Yes, this was introduced to surface those appenders contextually when a parent is selected, i.e. the intent is to add. The reason for the change is that people use the Columns block often to create layouts that feature entirely empty columns, f.ex. a 30% padding to the right of a main column. 

Note this appender wouldn't let you insert additional columns, though, this would be for inner blocks of the _Column_ block, singular.

**Comment by Mamaduka** [comment]
#20502 disabled appender for inner blocks, which allows adding additional columns when parent block is selected.

## Screenshot
![Image](https://github.com/user-attachments/assets/6f2d6710-6c18-45f9-9b27-5bacebf2799e)

**Comment by jasmussen** [comment]
Ah gotcha. Yes, in general that appender is not very intuitive, and often gets in the way. I wasn't involved in the context of removing that. It still seems like there's a way to improve the sibling inserter for this: in principle it's the same interface, the sibling inserter can just be visually more coherent by sitting in the gutter to the left or right of a columns block.

**Comment by Mamaduka** [comment]
Cross-linking a couple of related issues, props to @carolinan!

* https://github.com/WordPress/gutenberg/issues/30716
* https://github.com/WordPress/gutenberg/issues/30991
